### PR TITLE
C++ suggestions to remove some warnings and style

### PR DESCRIPTION
**Pull request by bolry**
Take what you like.

### CODE REVIEW

**Comment by bolry** [comment]
This  builds on the bugfix/cmake-install-issue-20240705 branch/PR.

File: src/wara_ps_client.cc [hunk 1]
```diff
@@ -10,39 +10,46 @@
...
-            {"levels",     {"sensor", "direct execution"}},
-            {"name",       "ljudkriget"},
-            {"rate",       heartbeat_interval.count() / 1000},
-            {"stamp",      (double) std::chrono::system_clock::now().time_since_epoch().count() / 1000.0},
-            {"type",       "HeartBeat"}};
+        {"agent-type", "surface"},
+        {"agent-uuid", kUUID},
+        {"levels",     {"sensor", "direct execution"}},
+        {"name",       "ljudkriget"},
+        {"rate",       duration_cast<milliseconds>(heartbeat_interval).count()},
```
Comment (bolry) [inline src/wara_ps_client.cc:42]: You likely want microseconds here.

File: src/wara_ps_client.cc [hunk 2]
```diff
@@ -10,39 +10,46 @@
...
 }
 
 std::string WaraPSClient::GenerateHeartBeatMessage() const {
     json j = {
-            {"agent-type", "surface"},
-            {"agent-uuid", kUUID},
-            {"levels",     {"sensor", "direct execution"}},
-            {"name",       "ljudkriget"},
-            {"rate",       heartbeat_interval.count() / 1000},
-            {"stamp",      (double) std::chrono::system_clock::now().time_since_epoch().count() / 1000.0},
```
Comment (bolry) [inline src/wara_ps_client.cc:36]: I guess this solution depends on system_clock returning a nanosecond resolution?

File: src/wara_ps_client.h [hunk 3]
```diff
@@ -50,16 +57,12 @@ class WaraPSClient {
     WaraPSClient(std::string name, std::string server_address);
 
     /**
-     * Deleted constructors to disallow copying and moving of the client
+     * Deleted constructors to disallow copying and assignment of the client
      */
-    WaraPSClient(WaraPSClient &&) = delete;
-
     WaraPSClient(const WaraPSClient &) = delete;
```
Comment (bolry) [inline src/wara_ps_client.h:62]: Declaring this will also automatically never declare the move constructor.

File: src/wara_ps_client.h [hunk 4]
```diff
@@ -50,16 +57,12 @@ class WaraPSClient {
...
 
     WaraPSClient &operator=(const WaraPSClient &) = delete;
```
Comment (bolry) [inline src/wara_ps_client.h:64]: Marking assignment operator as `= delete` will also automatically "remove" the move assignment operator.

File: src/wara_ps_client.cc (hunk 2 above)
Comment (Karlkorv) [reply src/wara_ps_client.cc:36]: The heartbeat message is based on this format: https://api.docs.waraps.org/#/agent_communication/topics/heartbeat
Which is supposed to be in seconds.

File: src/wara_ps_client.cc (hunk 1 above)
Comment (Karlkorv) [reply src/wara_ps_client.cc:42]: Isn't UNIX Epoch time usually in milliseconds? I might be misremembering.

File: src/wara_ps_client.h (hunk 3 above)
Comment (Karlkorv) [reply src/wara_ps_client.h:62]: Very nice! Didn't know this.

**Review by Karlkorv — APPROVED** [review]
Looks very good to me.

**Pull request merged**